src/bin/playground.cpp

```diff
@@ -408,7 +417,55 @@ file_header read_file_header(std::string filename) {
   return file_header;
 }
 
+void prepare_filestream(const char* fname) {
+  std::remove(fname);
+  FILESTREAM.open(fname, std::ios::out | std::ios::binary | std::ios::app);
+}
+
+void end_filestream() {
+  FILESTREAM.close();
+}
+
+void persist_chunks_to_disk(std::vector<std::shared_ptr<Chunk>> chunks) {
+  file_lock.acquire();
+
+  auto chunk_segment_offset_ends = std::vector<std::vector<uint32_t>>(CHUNK_COUNT);
+  auto chunk_offset_ends = std::array<uint32_t, CHUNK_COUNT>();
+  auto chunk_ids = std::array<uint32_t, CHUNK_COUNT>();
+
+  auto offset = uint32_t{sizeof(file_header)};
```

Comment:
preferably use `FILE_HEADER_BYTES` for this

Reply:
Will be part of next commit :)